src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Generated/Recipe.cs

```diff
@@ -238,6 +242,18 @@ private void ConfigureBeanstalkEnvironment(Configuration settings, string beanst
                    }
                 };
 
+            if (newDeployment ||
```

Comment:
if `newDeployment` is true, then the `if` statement is true without checking the 2nd condition.
If `newDeployment` is false, the 2nd condition will also be false since you are using &&

Reply:
The second condition was supposed to be `!newDeployment` for if doing a redeployment then skip if disable IMDSv1 is set to default. I had done some renaming refactoring where the variable used to be called `performRedeployment` and switched the naming to be `newDeployment` thinking that looks cleaner but forgot to update the inverse logic here.